datalad/downloaders/providers.py

```diff
@@ -439,12 +525,13 @@ def enter_new(self, url=None, auth_types=[]):
             title="Credential",
             text="What type of credential should be used?",
             choices=sorted(CREDENTIAL_TYPES),
-            default=getattr(authenticator_class, 'DEFAULT_CREDENTIAL_TYPE')
+            default=credential_type or getattr(authenticator_class,
+                                               'DEFAULT_CREDENTIAL_TYPE')
         )
 
-        # Just create a configuration file and reload the thing
-        if not path.lexists(providers_user_dir):
-            os.makedirs(providers_user_dir)
+        # TODO: This string is duplicated in self._store_new ATM. However,
+        #       that entire business of dealing with these config files should
+        #       go into a more sophisticated config parser/writer
```

Comment:
Why not define the template in a variable and use it twice? Seems to improve maintainability without requiring a config syntax/parser change.

Reply:
Honestly forgot why I decided against it. Will do (or re-discover why).